presto-main/src/main/java/io/prestosql/metadata/MetadataManager.java

```diff
@@ -493,6 +493,13 @@ public ColumnMetadata getColumnMetadata(Session session, TableHandle tableHandle
     {
         requireNonNull(prefix, "prefix is null");
 
+        Optional<QualifiedObjectName> objectName = prefix.asQualifiedObjectName();
+        if (objectName.isPresent()) {
+            return getTableHandle(session, objectName.get())
```

Comment:
But table itself could be also complicated, like having 100 or 1000 columns.


In case of JDBC both methods map to `io.prestosql.plugin.jdbc.BaseJdbcClient#getTables`. However in case of `QualifiedObjectName` it gets better predicate so your version should be faster.

Reply:
> But table itself could be also complicated, like having 100 or 1000 columns.

Good point. Maybe in the future we should have an additional `ConnectorMetadata` method or make `ConnectorMetadata#listTables` accept table name.

As you have checked, for JDBC it should improve performance via better predicate